onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -411,61 +641,104 @@ def _infer_Reshape(self, node):
                                                   get_shape_from_sympy_shape(new_sympy_shape)))
         self._pass_on_sympy_data(node)
 
+    def _infer_Scan(self, node):
+        subgraph = get_attribute(node, 'body')
+        num_scan_inputs = get_attribute(node, 'num_scan_inputs')
+        scan_input_axes = get_attribute(node, 'scan_input_axes', [0]*num_scan_inputs)
+        num_scan_states = len(node.input) - num_scan_inputs
```

Comment:
Negative axis values are supported here and in the output axes. Also there are a lot more ops that explicitly support negative axis following this PR: [https://github.com/onnx/onnx/pull/2281](https://github.com/onnx/onnx/pull/2281)